armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -359,6 +370,74 @@ def _getNucTempHelper(self):
             nv += nvBlock * wt
         return nvt, nv
 
+    def _getAverageComponentNumberDensities(self, compIndex):
+        """
+        Get weighted average number densities of a component in the collection.
+
+        Returns
+        -------
+        numberDensities : dict
+            nucName, ndens data (atoms/bn-cm)
+        """
+        nuclides = self.allNuclidesInProblem
```

Comment:
I think I was just following the pattern used for the `_getNucTempHelper` methods; I don't know if I had a particular reason when I initially did it.

Looking at it now, I could come up with a possible reason: you need all of the nuclides in all of the candidate blocks. It's possible that some blocks would have nuclides that others don't have. So you'd have to iterate over every block and update the list of nuclides.

This could still be done, and maybe it's better to do it that way. A little more code than the single line here, but it could be more efficient in some cases.